tests/votinghistory.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+import sys
+import os
+import io
+
+import e2e_args
+import getpass
+import time
+import logging
+import multiprocessing
+from random import seed
+import infra.ccf
+import infra.proc
+import infra.jsonrpc
+import infra.remote
+import json
+import ledger
+import msgpack
+
+import cryptography.x509
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.backends import default_backend
+
+
+def verify_sig(raw_cert, sig, req):
+    cert = cryptography.x509.load_der_x509_certificate(
+        raw_cert, backend=default_backend()
+    )
+    pub_key = cert.public_key()
+    hash_alg = ec.ECDSA(cert.signature_hash_algorithm)
+    pub_key.verify(sig, req, hash_alg)
+
+
+def run(args):
+    hosts = ["localhost", "localhost"]
+
+    ledger_filename = None
+
+    with infra.ccf.network(
+        hosts, args.build_dir, args.debug_nodes, args.perf_nodes, pdb=args.pdb
+    ) as network:
+        primary, others = network.start_and_join(args)
+
+        # create a lua query file to change a member state to accepted
+        with open("query.lua", "w") as qfile:
```

Comment:
Could we re-use the script in `memberclient.py` instead?

Reply:
Sorry, this shouldn't be here at all.